### PR TITLE
chore(functions): update cloud functions to specify build SA

### PR DESCRIPTION
**Pull request by eeaton**
to address https://github.com/terraform-google-modules/terraform-example-foundation/issues/1269.

I haven't been able to test it because our CI test org was not impacted by the default changes to cloud build SA, and has not encountered the permissions error.

### CODE REVIEW

File: 1-org/envs/shared/remote.tf [hunk 1]
```diff
@@ -19,18 +19,19 @@
...
+  project_prefix                                    = data.terraform_remote_state.bootstrap.outputs.common_config.project_prefix
+  folder_prefix                                     = data.terraform_remote_state.bootstrap.outputs.common_config.folder_prefix
+  networks_step_terraform_service_account_email     = data.terraform_remote_state.bootstrap.outputs.networks_step_terraform_service_account_email
+  org_step_terraform_service_account_email          = data.terraform_remote_state.bootstrap.outputs.organization_step_terraform_service_account_email
+  bootstrap_folder_name                             = data.terraform_remote_state.bootstrap.outputs.common_config.bootstrap_folder_name
+  cloud_build_private_worker_pool_id                = try(data.terraform_remote_state.bootstrap.outputs.cloud_build_private_worker_pool_id, "")
+  required_groups                                   = data.terraform_remote_state.bootstrap.outputs.required_groups
+  organization_step_terraform_service_account_email = data.terraform_remote_state.bootstrap.outputs.organization_step_terraform_service_account_email
```
Comment (daniel-cit) [inline 1-org/envs/shared/remote.tf:34]: duplicated with 
```
org_step_terraform_service_account_email          = data.terraform_remote_state.bootstrap.outputs.organization_step_terraform_service_account_email
```

File: 1-org/envs/shared/cai_monitoring.tf [hunk 2]
```diff
@@ -17,8 +17,9 @@
...
 
-  org_id          = local.org_id
-  billing_account = local.billing_account
-  project_id      = module.scc_notifications.project_id
-  location        = local.default_region
+  org_id                = local.org_id
+  billing_account       = local.billing_account
+  project_id            = module.scc_notifications.project_id
+  location              = local.default_region
+  build_service_account = local.organization_step_terraform_service_account_email
```
Comment (daniel-cit) [inline 1-org/envs/shared/cai_monitoring.tf:24]: `build_service_account` need to be in the full format: `projects/{{project}}/serviceAccounts/{{email}}`
like  `"projects/${local.seed_project_id}/serviceAccounts/${local.organization_step_terraform_service_account_email}"`
also reading the `seed_project_id` from the `0-bootstrap` remote state

**Review by daniel-cit — CHANGES_REQUESTED** [review]
the build service account need to be in the full format `projects/{{project}}/serviceAccounts/{{email}}`

**Comment by daniel-cit** [comment]
maybe a new service account created in the SCC project would provide a better separation of concerns. 
the org step service account has some org level roles that would not be need for the cloud build process of the CAI function

File: 1-org/envs/shared/cai_monitoring.tf (hunk 2 above)
Comment (eeaton) [reply 1-org/envs/shared/cai_monitoring.tf:24]: will revise format in next commit, thanks

File: 1-org/envs/shared/remote.tf (hunk 1 above)
Comment (eeaton) [reply 1-org/envs/shared/remote.tf:34]: will remove the duplicate local var in next commit

**Comment by daniel-cit** [comment]
@eeaton 
A quick ix for the build failure would be changing

from
```
    runtime_env_variables = {
      ROLES     = join(",", var.roles_to_monitor)
      SOURCE_ID = google_scc_source.cai_monitoring.id
    }
```

to
```
    runtime_env_variables = {
      ROLES            = join(",", var.roles_to_monitor)
      SOURCE_ID        = google_scc_source.cai_monitoring.id
      LOG_EXECUTION_ID = "true"
    }
```

here 
https://github.com/eeaton/terraform-example-foundation/blob/eeaton-fix-cloudfuncitons-build-default-sa/1-org/modules/cai-monitoring/main.tf#L164C5-L167C6


Change tested locally.
It does not list changes in the plan after this fix

**Comment by eeaton** [comment]
CI tests are failing due to a change unrelated to the content of this PR, same issue as addressed in [this comment](https://github.com/terraform-google-modules/terraform-example-foundation/pull/1210#issuecomment-2197195087). Once PR 1210 is merged, I expect the tests will pass

**Review by daniel-cit — APPROVED** [review]
LGTM

**Comment by daniel-cit** [comment]
@eeaton @apeabody The build is green 💚

**Review by apeabody — APPROVED** [review]
Thanks @eeaton!

reviewed/approved by @daniel-cit

**Pull request merged**